02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/08 Tslearn/04 Build Models.html

```diff
@@ -0,0 +1,12 @@
+<p>In this example, a time-series clustering model will be trained. Instead of using real-time comparison, we could apply a technique call Dynamic Time Wrapping (DTW) with Barycenter Averaging (DBA). Intuitively, it is a technique of averaging a few time-series into a single one without losing much of their information. Since not all time-series would move efficiently like in ideal EMH assumption, this would allow similarity analysis of different time-series with sticky lags. Check the technical details from <a href="https://tslearn.readthedocs.io/en/stable/user_guide/dtw.html">tslearn documentation page</a>.</p>
```

Comment:
> In this example, train a model that clusters the universe of Equities into distinct groups and then allocate an equal portion of the portfolio to each cluster. To cluster the securities, instead of using a real-time comparison, apply Dynamic Time Wrapping Barycenter Averaging (DBA) to their historical prices and then run a k-means clustering algorithm. DBA is a technique of averaging a few time-series into a single one without losing much of their information. Since not all time-series move efficiently like in ideal EMH assumption, this technique allows similarity analysis of different time-series with sticky lags. The following image shows a visualization of the process. For more information about the technical details, see Dynamic Time Warping in the <a href="https://tslearn.readthedocs.io/en/stable/user_guide/dtw.html">tslearn documentation</a>.